app/models/order.rb

```diff
@@ -17,4 +13,26 @@ def grandtotal
   def total_item_quantity
     item_orders.sum('quantity')
   end
+
+  def cancel_process
+    update(status: 1)
+    item_orders.each do |item_order|
+      item_order.update(status: 0)
+      item_order.item.update(inventory: (item_order.item.inventory + item_order.quantity))
+    end
+  end
+
+  def fulfill
+    item_orders.each do |item_order|
+      item_order.update(status: 1)
+    end
+    update(status: 2)
+  end
+
+  def ship_and_fulfill
+    item_orders.each do |item_order|
+      item_order.update(status: 1)
+    end
+    update(status: 3)
+  end
```

Comment:
Methods explained above in the controllers where they are deployed.  In cancel_process the items ar incremented back what was in that order.